pkg/router/template/plugin.go

```diff
@@ -291,6 +292,18 @@ func createRouterEndpoints(endpoints *kapi.Endpoints, excludeUDP bool, lookupSvc
 
 	out := make([]Endpoint, 0, len(endpoints.Subsets)*4)
 
+	// Return address as "[<address>]" if an IPv6 address,
+	// otherwise address is returned unadorned.
+	formatIPAddr := func(address string) string {
+		if ip := net.ParseIP(address); ip != nil {
+			// TODO(frobware): is counting ":" strictly necessary here?
+			if ip.To4() == nil && strings.Count(address, ":") >= 2 {
+				return "[" + address + "]"
```

Comment:
I was wondering what would happen if `address` were already wrapped in brackets (for example, what does `formatIPAddr("[::1]")` return?).  What happens is that `net.ParseIP` returns nil, so execution continues to the `return address` statement below (so `formatIPAddr("[::1]")` returns `"[::1]"`).  So the logic is a little non-obvious, but it seems to be correct.